NICE.Identity.Management/ClientApp/src/views/EditOrganisation/EditOrganisation.tsx

```diff
@@ -0,0 +1,312 @@
+
+		if (formHasNotChanged) {
+			this.setState({
+				isSaveButtonLoading: false,
+				hasSubmitted: true,
+			});
+			return false;
+		}
+
+		if (this.typingTimer) {
+			clearTimeout(this.typingTimer);
+			await this.checkOrgName(formName);
+		}
+
+		const { fetchedOrgNameFound } = this.state;
+
+		if (!form.checkValidity() || validationError || fetchedOrgNameFound) {
+			const validationErrorMessage = this.toggleValidationMessage(
+				fetchedOrgNameFound,
+				formName,
+			);
+			this.setState({
+				validationError: true,
+				validationErrorMessage,
+				isSaveButtonLoading: false,
+			});
+			return false;
+		}
+
+		const fetchOptions = {
+			method: "PATCH",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify({
+				name: formName,
+			}),
+		};
+
+		const organisation = await fetchData(
+			Endpoints.organisation(this.props.match.params.id),
+			fetchOptions,
+		);
+
+		if (isDataError(organisation)) {
+			this.setState({ error: organisation });
+		}
+
+		this.setState({
+			isSaveButtonLoading: false,
+			hasSubmitted: true,
+		});
+	};
+
+	handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+		const formElement = e.target;
+		const formName = formElement.value;
+		const isNowValid = formElement.validity.valid;
+		let { validationError } = this.state;
+
+		if (isNowValid) {
+			validationError = false;
+		}
+
+		clearTimeout(this.typingTimer);
+		this.typingTimer = setTimeout(this.checkOrgName, 1000, formName.trim());
+
+		this.setState({
+			formName,
+			validationError,
+		});
+	};
+
+	handleBlur = async (
+		e: React.ChangeEvent<HTMLInputElement>,
+	): Promise<void> => {
+		const formElement = e.target;
+		const formName = this.state.formName.trim();
+
+		if (this.typingTimer) {
+			clearTimeout(this.typingTimer);
+			await this.checkOrgName(formName);
+		}
+
+		const { fetchedOrgNameFound } = this.state;
+
+		const isNowValid = formElement.validity.valid
+			? !fetchedOrgNameFound
+			: false;
+
+		const validationErrorMessage = this.toggleValidationMessage(
+			fetchedOrgNameFound,
+			formName,
+		);
+
+		this.setState({
+			validationError: !isNowValid,
+			validationErrorMessage,
+			formName,
+		});
+	};
+
+	render(): JSX.Element {
+		const {
+			organisation,
+			formName,
+			isLoading,
+			hasSubmitted,
+			error,
+			isSaveButtonLoading,
+			validationError,
+			validationErrorMessage,
+		} = this.state;
+		const organisationId = this.props.match.params.id;
+
+		return (
+			<>
+				<Breadcrumbs>
+					<Breadcrumb to="/overview" elementType={Link}>
+						Administration
+					</Breadcrumb>
+					<Breadcrumb to="/organisations" elementType={Link}>
+						Organisations
+					</Breadcrumb>
+					<Breadcrumb
+						to={`/organisations/${organisationId}`}
+						elementType={Link}
+					>
+						{isLoading ? "Loading organisation details" : organisation.name}
+					</Breadcrumb>
+					<Breadcrumb>Edit organisation</Breadcrumb>
+				</Breadcrumbs>
+
+				{!error ? (
```

Comment:
Would this be more readable if you just reversed the render blocks in the ternary so you didn't have to qualify it with 'not'

Reply:
I think it was more so the non-error stuff took precedence above the error, but I can see your point.